broker/src/test/java/io/camunda/zeebe/broker/bootstrap/EmbeddedGatewayServiceStepTest.java

```diff
@@ -0,0 +1,159 @@
+import static org.mockito.Mockito.verify;
+
+import io.camunda.zeebe.broker.SpringBrokerBridge;
+import io.camunda.zeebe.broker.clustering.ClusterServicesImpl;
+import io.camunda.zeebe.broker.system.EmbeddedGatewayService;
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.monitoring.BrokerHealthCheckService;
+import io.camunda.zeebe.protocol.impl.encoding.BrokerInfo;
+import io.camunda.zeebe.test.util.socket.SocketUtil;
+import io.camunda.zeebe.util.sched.ActorScheduler;
+import io.camunda.zeebe.util.sched.TestConcurrencyControl;
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public class EmbeddedGatewayServiceStepTest {
+  private static final TestConcurrencyControl CONCURRENCY_CONTROL = new TestConcurrencyControl();
+  private static final BrokerCfg TEST_BROKER_CONFIG = new BrokerCfg();
+
+  static {
+    final var networkCfg = TEST_BROKER_CONFIG.getGateway().getNetwork();
+    networkCfg.setHost("localhost");
+  }
+
+  private final EmbeddedGatewayServiceStep sut = new EmbeddedGatewayServiceStep();
+  private BrokerStartupContextImpl testBrokerStartupContext;
+
+  @Test
+  void shouldHaveDescriptiveName() {
+    // when
+    final var actual = sut.getName();
+
+    // then
+    assertThat(actual).isSameAs("Embedded Gateway");
+  }
+
+  @Nested
+  class StartupBehavior {
+
+    private ActorFuture<BrokerStartupContext> startupFuture;
+    private ActorScheduler actorScheduler;
+
+    @BeforeEach
+    void setUp() {
+      actorScheduler = ActorScheduler.newActorScheduler().build();
+      actorScheduler.start();
+      startupFuture = CONCURRENCY_CONTROL.createFuture();
+
+      testBrokerStartupContext =
+          new BrokerStartupContextImpl(
+              mock(BrokerInfo.class),
+              TEST_BROKER_CONFIG,
+              mock(SpringBrokerBridge.class),
+              actorScheduler,
+              mock(BrokerHealthCheckService.class));
+
+      testBrokerStartupContext.setClusterServices(
+          mock(ClusterServicesImpl.class, RETURNS_DEEP_STUBS));
+
+      final var port = SocketUtil.getNextAddress().getPort();
+      final var commandApiCfg = TEST_BROKER_CONFIG.getGateway().getNetwork();
+      commandApiCfg.setPort(port);
+    }
+
+    @AfterEach
+    void tearDown() {
+      final var embeddedGatewayService = testBrokerStartupContext.getEmbeddedGatewayService();
+      if (embeddedGatewayService != null) {
+        embeddedGatewayService.close();
+      }
+      actorScheduler.stop();
+    }
+
+    @Test
+    void shouldCompleteFuture() {
+      // when
+      sut.startupInternal(testBrokerStartupContext, CONCURRENCY_CONTROL, startupFuture);
+      await().until(startupFuture::isDone);
+
+      // then
+      assertThat(startupFuture.isCompletedExceptionally()).isFalse();
```

Comment:
:wrench: IIRC, AssertJ has future specific assertions, so you can do things like:

```java
assertThat(startupFuture).succeedsWithin(Duration.ofSeconds(10)).isEqualTo(testBrokerStartupContext);
```

Or when you expect it to fail:

```java
assertThat(startupFuture).failsWithin(Duration.ofSeconds(5)).isInstanceOf(ExecutionException.class).havingRootCause(NullPointerException.class);
```

Reply:
I will do this in a general code cleanup for all tests